.github/actions/publish/action.yml

```diff
@@ -29,17 +30,42 @@ runs:
       with:
         platforms: linux/amd64,linux/arm64/v8,linux/arm/v7,linux/386
     - name: Login to Docker
+      shell: bash
+      run: echo $DOCKER_TOKEN | docker login --username $DOCKER_USERNAME --password-stdin
+
+    - name: Set up goreleaser
       shell: bash
       run: |
-        echo $DOCKER_TOKEN | docker login --username $DOCKER_USERNAME --password-stdin
+        CONTAINER_ID="$(
+          docker run --detach \
+            --volume "$PWD:$PWD" \
+            --entrypoint tail \
+            ghcr.io/goreleaser/goreleaser-cross:latest \
+              -f /dev/null
+        )"
+        docker exec --tty "$CONTAINER_ID" dpkg --add-architecture i386
+        docker exec --tty "$CONTAINER_ID" apt-get update
+        docker exec --tty "$CONTAINER_ID" apt-get install --no-install-recommends -y -q crossbuild-essential-i386
+        docker exec --tty "$CONTAINER_ID" git config --global --add safe.directory '*'
+        echo "CONTAINER_ID=$CONTAINER_ID" >> "$GITHUB_ENV"
+
     - name: Run Goreleaser
-      uses: goreleaser/goreleaser-action@v5
-      with:
-        version: latest
-        args: release ${{ inputs.dry-run == 'true' && '--skip=publish' || '' }} --config .goreleaser.yaml
+      shell: bash
+      run: docker exec
+        --env GITHUB_TOKEN
+        --env HOMEBREW_DEPLOY_KEY
+        --workdir "$PWD"
+        --tty
+        "$CONTAINER_ID"
+        goreleaser release ${{ inputs.dry-run == 'true' && '--skip=publish' || '' }} --config .goreleaser.yaml
       env:
         GITHUB_TOKEN: ${{ inputs.token }}
         HOMEBREW_DEPLOY_KEY: ${{ inputs.homebrew-gh-secret }}
+    - name: Upload assets
```

Comment:
Has it been verified that the uploaded results exactly match (or are expected) compared to previous releases, since the `goreleaser-action` is no longer being used?

Reply:
We're still using the goreleaser action. We're just also uploading the artifacts that were generated alongside the action run, so that it's easier to test dry runs.